### DISCUSSION TITLE
Instructions (voice, visual) mismatch with own route on map

### DISCUSSION THREAD

**Discussion by Tukcedo** [osmandapp/OsmAnd#11356] [General]
Hi All!

I'm a relatively new user, but I noticed something odd during navigation. When I create my own route by adding and dragging waypoints onto the map, there seems to be a mismatch between the voice and visual instructions to the point where the instructions seem totally unrelated (or at the very least, not synchronised) to the actual route shown on the map. I have "car" selected and during the track creation process, the route is properly following the roads after I add a waypoint, resulting in a route map that is (largely, see below) correct. 

- However, while driving I often get the instruction to go right whereas the map (correctly) shows a left or I'm asked to turn into some non-existing road. Not surprisingly, the instructions often use the wrong street name, whereas the map is correct. 
- The time needed to complete the route is way off: for something like 80k (again, for a car) it will start off with something like 40 hours but getting a little closer to something more realistic as we progress along the route
- Often, there seem to be points the navigation wants to go to that are maybe 100m or so off the route, in which case even the map shows these spurious "detours". Most often these occur in intersections like T-junctions. So navigating on the map, they're easy enough to avoid as you can usually see where the correct route goes.

Strangely enough, routes that I download (I've tried primarily those from the web site of the Dutch tourist organisation ANWB) seem to do much better at these things than the routes I create myself.

Am I doing something wrong? Are there any settings I need to change?

Thanks!

Mich.

Tablet: Samsung Tab A (SM-P550), Android v7.1.1, kernel 3.10.49-13808168
OsmAnd: 3.9.10 (free)

**Comment by vshcherb** [comment]
Quite late response, but it could be both bugs and misconfiguration. It's quite hard to say without actual route screenshots, so we could try to reproduce the issue.

**Comment by MatthijsDe** [comment]
Hi all,
I have exactly the same mismatch between voice and visual instuctions at times. Not only with routes created myself but also with imported GPX files.

**Reply by MatthijsDe** [reply]
Just an example of such a mismatch at a roundabout:
![02A2A43B-9C10-439A-964B-5CF1536254E4](https://user-images.githubusercontent.com/86867300/124391187-b6f90680-dcef-11eb-8bfa-fb96d2de2e5f.jpeg)

Visual guidance says take second street, voice says take third street.

**Reply by barteis** [reply]
This is probably a stretch, but OSM shows two concentric roundabouts (though I don't see that from the aerial imagery). Could that confuse navigation?  If you go from the outer roundabout to the inner one, does that count as the first junction?

**Comment by tommycrock** [comment]
If anyone wants to investigate this the relevant roundabout seems to be
https://www.openstreetmap.org/way/6800697#map=19/52.29672/6.52178

**Reply by vshcherb** [reply]
Just tested on newly generated map via MapCreator, correct - 2nd exit.
